orbstation/code/antagonists/traitor/items/booze_grenade.dm

```diff
@@ -0,0 +1,56 @@
+#define BOOZENADE_CAPACITY 50
+
+/obj/item/grenade/boozenade
+	name = "booze grenade"
+	desc = "A hefty grenade casing with a nozzle for pouring in ethanol products. When activated, it will disperse said \
+		products as a dense cloud of smoke, rapidly intoxicating anyone who breathes it in."
+	icon = 'orbstation/icons/obj/weapons/weapons.dmi'
+	icon_state = "boozenade"
+
+/obj/item/grenade/boozenade/Initialize(mapload)
+	. = ..()
+	create_reagents(BOOZENADE_CAPACITY, AMOUNT_VISIBLE)
+
+//Special proc to only allow booze to be poured inside. Partly copied from smoke machine.
+/obj/item/grenade/boozenade/attackby(obj/item/attacking_item, mob/user, params)
+	if(is_reagent_container(attacking_item) && attacking_item.is_open_container())
```

Comment:
can you flip the check and early return here so that everything isnt nested in this check?